lib/firehose/long_poll_transport.coffee

```diff
@@ -108,7 +108,7 @@ class LongPollTransport extends Transport
 
     unless @_needToNotifyOfReconnect or @_stopRequestLoop
       @_needToNotifyOfReconnect = true
-      @config.disconnected()
+      @config.disconnected(jqXhr, status, error)
```

Comment:
I'm fine switching to just the error object if you think that's better. Should we pass the transport type in all the events (connected, disconnected, failed, error)?